aten/src/ATen/native/Linear.cpp

```diff
@@ -641,9 +642,26 @@ Tensor tensordot(const Tensor& input1, const Tensor& input2, IntArrayRef dims1,
 }
 
 Tensor &tensordot_out(const Tensor& input1, const Tensor& input2, IntArrayRef dims1, IntArrayRef dims2, Tensor& result) {
-  result.copy_(at::native::tensordot(input1, input2, dims1, dims2));
+  Tensor result_tmp = at::native::tensordot(input1, input2, dims1, dims2);
+  auto result_dtype = result_tmp.scalar_type();
+  auto output_tensor_dtype = result.scalar_type();
+  auto output_device = result.device();
+  auto input_device = result_tmp.device();
+  // check if the input & output tensors are on the same device.
+  TORCH_CHECK(
```

Comment:
I can include `LinearAlgebraUtils.h` & reuse checks from there, but doing so would increase the binary size a little more.
Personally, I don't have a preference, so I'm willing to go for the other approach as well.

Reply:
Looks OK to me

Reply:
Sorry, I had misread your comment on `canCast`.
Prima facie, it does seem like a good idea to include the `canCast` check in `resize_output`.

Reply:
A good idea to pursue in a separate PR, maybe. I wonder how easy it is to always know the computation dtype to pass to the resize_output check?

Reply:
I'd also take a look & would ping you after doing so.